ts/torch_handler/base_handler.py

```diff
@@ -25,46 +21,74 @@
 else:
     PROFILER_AVAILABLE = False
 
+
+logger = logging.getLogger(__name__)
+
+
 try:
     import torch_xla.core.xla_model as xm
 
-    TORCHXLA_AVAILABLE = True
+    XLA_AVAILABLE = True
 except ImportError as error:
-    TORCHXLA_AVAILABLE = False
+    XLA_AVAILABLE = False
 
 
-logger = logging.getLogger(__name__)
-
-
-# Possible values for backend in utils.py
-def check_pt2_enabled():
-    try:
-        import torch._dynamo
-
-        pt2_enabled = True
-        if torch.cuda.is_available():
-            # If Ampere enable tensor cores which will give better performance
-            # Ideally get yourself an A10G or A100 for optimal performance
-            if torch.cuda.get_device_capability() >= (8, 0):
-                torch.backends.cuda.matmul.allow_tf32 = True
-    except ImportError as error:
-        logger.warning(
-            "dynamo/inductor are not installed. \n For GPU please run pip3 install numpy --pre torch[dynamo] --force-reinstall --extra-index-url https://download.pytorch.org/whl/nightly/cu117 \n for CPU please run pip3 install --pre torch --extra-index-url https://download.pytorch.org/whl/nightly/cpu"
-        )
-        pt2_enabled = False
-    return pt2_enabled
+if packaging.version.parse(torch.__version__) >= packaging.version.parse("2.0.0"):
+    PT2_AVAILABLE = True
+    if torch.cuda.is_available():
+        # If Ampere enable tensor cores which will give better performance
+        # Ideally get yourself an A10G or A100 for optimal performance
+        if torch.cuda.get_device_capability() >= (8, 0):
+            torch.backends.cuda.matmul.allow_tf32 = True
```

Comment:
@msaroufim wondering if `torch.set_float32_matmul_precision('high')` will be set automatically, in my benchmark before I found in for example HF models it was not set automatically and setting it change the perf substantially.

Reply:
Yeah those are the same thing but I agree that warnings are suggesting matmul precision so will change

Reply:
I see,  it seems setting to HIGH will allow tf32 https://github.com/pytorch/pytorch/blob/d8d479c854c622a6ef21190b7e62c9a76f0ea2a7/aten/src/ATen/Context.cpp#L178-L184,